src/index.ts

```diff
@@ -3,146 +3,145 @@ import { context, getOctokit } from '@actions/github';
+
+            return await getCoverage(ignoreCollector, options!, true);
         }
+    );
+
+    if (baseCoverage) {
+        dataCollector.add(baseCoverage);
+    }
 
-        if (
-            jsonReport &&
-            isAnnotationEnabled(annotations, 'coverage') &&
-            headReport.summary
-        ) {
-            const coverageAnnotations = createCoverageAnnotations(jsonReport);
-            if (coverageAnnotations.length > 0) {
-                const coverage = headReport.summary.find(
-                    (value) => value.title === 'Statements'
-                )!.percentage;
-                try {
-                    await octokit.checks.create(
-                        formatCoverageAnnotations(
-                            !threshold || coverage > threshold,
-                            coverage,
-                            threshold!,
-                            coverageAnnotations
-                        )
-                    );
-                } catch (err) {
-                    console.error('Failed to create annotations', err);
-                }
+    const [isReportContentGenerated, reportContent] = await runStage(
+        'generateReportContent',
+        dataCollector,
+        async (skip) => {
+            if (!isInitialized) {
+                skip();
             }
+
+            return createReport(dataCollector, options?.workingDirectory);
         }
+    );
 
-        if (isInPR && baseReport && pull_request) {
+    await runStage('publishReport', dataCollector, async (skip) => {
+        if (!isReportContentGenerated) {
+            skip();
+        }
+
+        const octokit = getOctokit(options!.token);
+
+        if (isInPR) {
             await generatePRReport(
-                (icons as Record<string, Icons>)[iconType],
-                headReport,
-                baseReport,
-                threshold,
-                repo,
-                pull_request,
-                octokit,
-                workingDirectory,
-                customTitle
-            );
-        } else if (!isInPR) {
-            await generateCommitReport(
-                (icons as Record<string, Icons>)[iconType],
-                headReport,
-                threshold,
-                repo,
-                octokit,
-                workingDirectory,
-                customTitle
+                reportContent!,
+                options!.workingDirectory,
+                context.repo,
+                context.payload.pull_request!,
+                octokit
             );
         } else {
-            throw new Error(
-                'Something went wrong! Looks like action runs in PR, but report for the base branch or pull_request information is missing!'
-            );
+            await generateCommitReport(reportContent!, context.repo, octokit);
+        }
+    });
+
+    await runStage('failedTestsAnnotations', dataCollector, async (skip) => {
+        if (!isInitialized || isHeadCoverageGenerated) {
```

Comment:
isHeadCoverageGenerated will be true even if a test fails, so this step will be skipped.

Reply:
https://github.com/sidharthv96/jest-coverage-report-action-test/pull/1/checks?check_run_id=3451046926#step:3:276

<img width="376" alt="image" src="https://user-images.githubusercontent.com/10703445/131223607-119a37ed-cbce-47da-a2b2-1ab258dcf654.png">